src/noted-objects-analysis/types-parser.js

```diff
@@ -123,8 +123,12 @@ function getProperties(comment) {
 
   let pos = -1
   while ((pos = comment.indexOf('@property', pos + 1)) > -1) {
-    const propertyStr = comment.substring(pos, comment.indexOf('\n', pos));
-    const deconstruction = /\{(.*?:)?(.*?)\}\s*(\[[A-z0-9\_=\'\"\s]+\]|[A-z0-9\_=\'\"]+)\s*(.*)/g.exec(propertyStr);
+    const nextPropertyIndex = comment.indexOf('@property', pos + 1);
+    const propertyStr = comment
+      .substring(pos, nextPropertyIndex < 0 ? comment.length - 1 : nextPropertyIndex)
+      .replace(/\n\s*\*\s*/g, "\n * ") // reduce whitespace around *
+      .replace(/\s\*\s$/, ""); // remove last empty line
+    const deconstruction = /\{(.*?:)?(.*?)\}\s+(\[[A-z0-9\_=\'\"\s]+\]|[A-z0-9\_=\'\"]+)\s*([\s\S]*)/g.exec(propertyStr);
```

Comment:
This is a very convoluted regex but it works. Maybe we can simplify a bit by having something like this
```suggestion
    const deconstruction = /\{(.*?:)?(.*?)\}\s+(\[[\w=\'\"\s]+\]|[\w=\'\"]+)\s*([\s\S]*)/g.exec(propertyStr);
```

Reply:
In my view it really does not make much difference in improving understanding of the expression but it is shorter.

Reply:
That's fair; the one change it does make is that it includes `[a-zA-Z]` as opposed to what is currently written (according the regex checkers at regex101). Not sure how much of a difference that will make in reality.